README.md

```diff
@@ -1,86 +1,125 @@
 
-## Configuring with cmake
+#### Use cmake to build MIGraphX
 
-First create a build directory:
+In this approach, we need to install the dependencies, configure the cmake, and then build the source.
 
+##### Installing the dependencies
 
-```
-mkdir build; 
-cd build;
-```
+For convenience, the dependencies can be built automatically with rbuild as:
+```rbuild build -d depend --cxx=/opt/rocm/llvm/bin/clang++```, all the dependencies are in the 
+folder `depend`. 
 
-Next configure cmake. The hcc compiler is required to build the MIOpen backend:
+If you have sudo access, as an alternative to the rbuild command, you can install the dependencies just 
+like in the docker file by calling ```./tools/install_prereqs.sh.```
 
+(Note: By default, all dependencies are installed at the default location `/usr/local` 
+and are accessible by all users. For the default location, `sudo` is required to run the script.
+You can also specify a location at which the dependencies are installed with `./tools/install_prereqs.sh $your_loc`.)
 
-```
-CXX=/opt/rocm/llvm/bin/clang++ cmake ..
-```
+##### Building MIGraphX source and install libs
+
+With the above dependencies installed, we can build source as:
 
-If the dependencies from `install_deps.cmake` was installed to another directory, the `CMAKE_PREFIX_PATH` needs to be set to what `--prefix` was set to from `install_deps.cmake`:
+1) Go to the project folder and create a build directory:
 
 
 ```
-CXX=/opt/rocm/llvm/bin/clang++ cmake -DCMAKE_PREFIX_PATH=/some/dir ..
+mkdir build
+cd build
 ```
 
+2) Configure the cmake. If the dependencies are installed at the default location `/usr/local`, the command is:
 
-#### Changing the cmake configuration
-
-The configuration can be changed after running cmake by using `ccmake`:
-
-` ccmake .. ` **OR** `cmake-gui`: ` cmake-gui ..`
-
-## Building the library
-
-The library can be built, from the `build` directory using the 'Release' configuration:
-
-` cmake --build . --config Release ` **OR** ` make `
+```
+CXX=/opt/rocm/llvm/bin/clang++ cmake ..
+```
+Otherwise, you need to set `-DCMAKE_PREFIX_PATH=$your_loc` to configure the cmake. 
 
-And can be installed by using the 'install' target:
+3) Build MIGraphX source code
 
-` cmake --build . --config Release --target install ` **OR** ` make install `
+```
+make -j$(nproc)
+```
 
-This will install the library to the `CMAKE_INSTALL_PREFIX` path that was set.
+Correctness can be verified as:
 
-To build a debug version of the library, the cmake variable `CMAKE_BUILD_TYPE` can be set to `Debug`.
+```
+make -j$(nproc) check
+```
 
-` cmake -DCMAKE_BUILD_TYPE=Debug . `
+MIGraphX libs can be installed as:
 
-## Running the tests
+```
+make install
+```
 
-The tests can be run by using the 'check' target:
+### Calling MIGraphX APIs
+To use MIGraphX's C/C++ API in your cmake project, then set `CMAKE_PREFIX_PATH` to your rocm installation and then do 
+```
+find_package(migraphx)
+target_link_libraries(myApp migraphx::migraphx_c)
```

Comment:
In the C++ examples I've done so far, this needed to be migraphx::c, rather than migraphx::migraphx_c, to compile. @pfultz2 can you confirm whether this is the case?

Reply:
Yes, thats correct!